src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,54 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  static addAnimal(animal) {
+    this.alive.push(animal);
+  }
+
+  static removeAnimal(animal) {
+    const indexOfAnimal = this.alive.indexOf(animal);
+
+    if (indexOfAnimal !== -1) {
+      this.alive.splice(indexOfAnimal, 1);
+    }
+  }
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.addAnimal(this);
+  }
+
+  hide() {
+    this.hidden = this.hidden ? 0 : true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    Animal.addAnimal(this);
+  }
+  bite(herbivore) {
+    if (!herbivore.hidden && herbivore.constructor === Herbivore) {
```

Comment:
Use `instanceOf` here instead 
```suggestion
  bite(herbivore) {
    if (!herbivore.hidden && herbivore.constructor === Herbivore) {
```

Reply:
At first you should check on instanceOf and then on hidden property it's logically correct